circe/src/main/scala/org/http4s/circe/CirceInstances.scala

```diff
@@ -218,11 +214,40 @@ object CirceInstances {
 
   // Constant byte chunks for the stream as JSON array encoder.
 
-  private final val openBrace: Stream[fs2.Pure, Chunk[Byte]] =
-    Stream.emit(Chunk.singleton('['.toByte))
+  private def fromJsonToChunk(printer: Printer)(json: Json): Chunk[Byte] =
+    Chunk.byteBuffer(printer.printToByteBuffer(json))
 
-  private final val closeBrace: Stream[fs2.Pure, Chunk[Byte]] =
-    Stream.emit(Chunk.singleton(']'.toByte))
+  private def streamedJsonArray[F[_]](printer: Printer)(s: Stream[F, Json]): Stream[F, Byte] =
+    s.pull.uncons1.flatMap {
+      case None => Pull.done
+      case Some((hd, tl)) =>
+        Pull.output(
+          Chunk.concatBytes(Vector(CirceInstances.openBrace, fromJsonToChunk(printer)(hd)))
+        ) >> // Output First Json As Chunk with leading `[`
+          tl.repeatPull {
+            _.uncons.flatMap {
+              case None => Pull.pure(None)
+              case Some((hd, tl)) =>
+                val interspersed = {
+                  val bldr = Vector.newBuilder[Chunk[Byte]]
+                  bldr.sizeHint(hd.size * 2)
+                  hd.foreach { o =>
+                    bldr += CirceInstances.comma
+                    bldr += fromJsonToChunk(printer)(o)
+                  }
+                  Chunk.concatBytes(bldr.result())
+                }
+                Pull.output(interspersed) >> Pull.pure(Some(tl))
+            }
+          }.pull
+            .echo
+    }.stream ++ Stream.chunk(closeBrace)
```

Comment:
@ChristopherDavenport @rossabaker In the case of encoding an empty list using streams, I notice the `closeBrace` is printed always at then, but the `openBrace` is only printed if it has elements. If not, it will use `Pull.done` and the encoding will only be: `]`, right?